packages/webapp/src/inductions/api/eden-contract.ts

```diff
@@ -15,15 +15,16 @@ const INDEX_BY_INVITER = 3;
 const INDEX_BY_ENDORSER = 2;
 const INDEX_BY_INDUCTION = 3;
 
-export const getInductionWithEndorsements = async (
-    inductionId: string
-): Promise<
+export const getInductionWithEndorsements = async ({
+    queryKey,
+}: any): Promise<
```

Comment:
this one abstracts away how to use this function... not good...
before I knew that if  I want to "get induction with endorsements" I could simply pass the induction Id. now I have no idea what I should pass to this function... 
Also, I can't understand what is the purpose of this _key since we are not using it anywhere inside this function. I think this abstraction should live outside... eg:

```
export const useGetInductionWithEndorsements = (inductionId: string) =>
    useQuery(["induction", inductionId], () => getInductionWithEndorsements(inductionId), {
        // gets rid of bugs cause by refetching right after anchor/scatter signing
        refetchOnWindowFocus: false,
    });
```

With that we don't need to touch the getInductionWithEndorsements. And getInductionWithEndorsements can also be used outside react-query if we want to call it from our backend.

Reply:
This is more of the `react-query` way of doing this: https://react-query.tanstack.com/guides/query-functions#query-function-variables. But I don't feel strongly about it either way.

If we plan to use `getInductionWithEndorsements` outside of `react-query`, then switching it back is prob not a bad idea.

Reply:
Yeah I get it is something for react-query. The problem is when we mix a simple data fetch with something that does not belong to it, meaning, from outside. This would be the first function where we are leaking something from our "Data Management SDK" to one of our API calls. And this is not desirable at all, because if we try to use it in our backend, or if we want to get rid of React-Query or maybe reuse it with another lib (wrapping with graphql or whatsoever), we just can't. 

See our [Members Abstractions](https://github.com/eoscommunity/Eden/blob/main/packages/webapp/src/members/api/members.ts) or even the other api calls to our EOS tables in this very same file... They are pure abstractions of how we call external apis. It has zero knowledge about our fetch/state mechanism. Can be reused anywhere, with any library.